pkg/reconciler/autoscaling/hpa/resources/hpa_test.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"math"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/knative/pkg/ptr"
+	"github.com/knative/serving/pkg/apis/autoscaling"
+	"github.com/knative/serving/pkg/apis/autoscaling/v1alpha1"
+
+	autoscalingv2beta1 "k8s.io/api/autoscaling/v2beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	. "github.com/knative/serving/pkg/testing"
+)
+
+func TestMakeHPA(t *testing.T) {
+	cases := []struct {
+		name string
+		pa   *v1alpha1.PodAutoscaler
+		want *autoscalingv2beta1.HorizontalPodAutoscaler
+	}{{
+		name: "defaults",
+		pa:   pa(),
+		want: hpa(),
+	}, {
+		name: "with lower bound",
+		pa:   pa(WithLowerScaleBound(5)),
+		want: hpa(withMinReplicas(5), withAnnotationValue(autoscaling.MinScaleAnnotationKey, "5")),
+	}, {
+		name: "with upper bound",
+		pa:   pa(WithUpperScaleBound(5)),
+		want: hpa(withMaxReplicas(5), withAnnotationValue(autoscaling.MaxScaleAnnotationKey, "5")),
+	}, {
+		name: "with an actual target",
+		pa:   pa(WithTargetAnnotation("50"), WithMetricAnnotation(autoscaling.CPU)),
+		want: hpa(
+			withAnnotationValue(autoscaling.MetricAnnotationKey, autoscaling.CPU),
+			withAnnotationValue(autoscaling.TargetAnnotationKey, "50"),
+			withMetric(autoscalingv2beta1.MetricSpec{
+				Type: autoscalingv2beta1.ResourceMetricSourceType,
+				Resource: &autoscalingv2beta1.ResourceMetricSource{
+					Name:                     corev1.ResourceCPU,
+					TargetAverageUtilization: ptr.Int32(50),
+				},
+			})),
+	}}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			if diff := cmp.Diff(tc.want, MakeHPA(tc.pa)); diff != "" {
+				t.Errorf("%q (-want, +got):\n%v", tc.name, diff)
```

Comment:
```suggestion
				t.Errorf("%s (-want, +got):\n%v", tc.name, diff)
```
you don't want your test name quoted :D